opendata-swiss/ckanext-geocat
# Fix tests

Test changes:

- fix permission of `test_harvester.py` so that those tests will run
- replace assertion that was mistakenly removed
- replace httpretty library with requests-mock: httpretty was not mocking requests correctly
- remove monkey-patching that was required due to a bug in httpretty
- update mocked responses from geocat to match the current behaviour
- use a custom solr schema for testing: this was necessary so that the field `identifier` would be saved onto datasets, and could be searched for during harvesting
- purge harvest queues and clear test datasets/harvesters from Solr during test setup and teardown
- fix test that expected behaviour that has since been fixed: existing datasets are now updated on import, rather than new datasets being created
- add test that the correct dataset is deleted, when it has been deleted at the source
- add local testing config to `test.ini` (commented out)

Harvester changes:

- use the dataset schema defined by applicable plugin, not the ckan default schema, to save datasets
- log errors that might occur while finding all datasets for an organization due to datasets being saved with the wrong schema. If this happens, return all valid datasets for the organization, not just a subset as happened previously. (This is unlikely in practice, but arose during testing because an incorrect dataset schema was used.)
- save a harvest error if there is a different error while finding all datasets for an organization
